File: backend/schemas/smtp_settings.py

from pydantic import BaseModel, Field


class SMTPSettingsResponse(BaseModel):
    """Current SMTP operational settings."""

    SMTP_DEFAULT_TIMEOUT: int = Field(..., ge=1)
    SMTP_RATE_LIMIT_PER_HOUR: int = Field(..., ge=1)
    SMTP_MAX_RETRIES: int = Field(..., ge=0)
    SMTP_MAX_DELAY: int = Field(..., ge=0)
    # Optional throttling extensions
    PER_DOMAIN_LIMITS: dict[str, int] | None = None  # domain -> msgs/hour
    WARMUP_PLAN: dict | None = None  # plan definition


class SMTPSettingsUpdate(BaseModel):
    """Payload for updating SMTP settings."""

    SMTP_DEFAULT_TIMEOUT: int | None = Field(None, ge=1)
    SMTP_RATE_LIMIT_PER_HOUR: int | None = Field(None, ge=1)
    SMTP_MAX_RETRIES: int | None = Field(None, ge=0)
    SMTP_MAX_DELAY: int | None = Field(None, ge=0)
    # Optional throttling extensions (applied if provided)
    PER_DOMAIN_LIMITS: dict[str, int] | None = None
    WARMUP_PLAN: dict | None = None

File: frontend/src/api/smtp-settings.ts
import { apiClient } from '@/http/stable-api-client'

export interface SmtpSettingsResponse {
    SMTP_DEFAULT_TIMEOUT: number
    SMTP_RATE_LIMIT_PER_HOUR: number
    SMTP_MAX_RETRIES: number
    SMTP_MAX_DELAY: number
}

export interface SmtpSettingsUpdate {
    SMTP_DEFAULT_TIMEOUT?: number
    SMTP_RATE_LIMIT_PER_HOUR?: number
    SMTP_MAX_RETRIES?: number
    SMTP_MAX_DELAY?: number
}

export async function getSmtpSettings(): Promise<SmtpSettingsResponse> {
    return apiClient.get<SmtpSettingsResponse>('/api/v1/smtp-settings/settings')
}

export async function updateSmtpSettings(payload: SmtpSettingsUpdate): Promise<SmtpSettingsResponse> {
    return apiClient.put<SmtpSettingsResponse>('/api/v1/smtp-settings/settings', payload)
}

